cluster-autoscaler/cloudprovider/gce/autoscaling_gce_client_test.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gce
+
+import (
+	"net/http"
+	"testing"
+
+	test_util "k8s.io/autoscaler/cluster-autoscaler/utils/test"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	gce_api "google.golang.org/api/compute/v1"
+)
+
+func newTestAutoscalingGceClient(t *testing.T, projectId, url string) *autoscalingGceClientV1 {
+	client := &http.Client{}
+	gceClient, err := NewAutoscalingGceClientV1(client, projectId)
+	if !assert.NoError(t, err) {
+		t.Fatalf("fatal error: %v", err)
+	}
+	gceClient.gceService.BasePath = url
+	return gceClient
+}
+
+func TestWaitForOp(t *testing.T) {
+	server := test_util.NewHttpServerMock()
+	defer server.Close()
+	g := newTestAutoscalingGceClient(t, "project1", server.URL)
+	server.On("handle", "/project1/zones/us-central1-b/operations/operation-1505728466148-d16f5197").Return(operationRunningResponse).Times(3)
+	server.On("handle", "/project1/zones/us-central1-b/operations/operation-1505728466148-d16f5197").Return(operationDoneResponse).Once()
+
+	operation := &gce_api.Operation{Name: "operation-1505728466148-d16f5197"}
+
+	err := g.waitForOp(operation, projectId, zoneB)
```

Comment:
I think unittests tests doing sleeps inside are a big anti-pattern. If UT takes more than a few seconds to run people just stop running them. I know in this case it's just 300ms, but I'd rather avoid it on principle (also because we may have more tests calling waitForOp internally in future).

Can we just make operationPollInterval a variable on autoscalingGceClientV1 (defaulting to current 100ms) and override it in tests?

Reply:
I just realized this is just copy-pasted. Still, I think applying my comment would be a nice improvement.

Reply:
Done, and added another tests case (timeout). I hope to move from GceManager tests that require waiting for operation each time to ones using fake client, but for now changing this brought down time needed for running unit tests by about 0.5s:

Before:
$ go clean -testcache && go test ./...
ok  	k8s.io/autoscaler/cluster-autoscaler/cloudprovider/gce	1.260s

After:
$ go clean -testcache && go test ./...
ok  	k8s.io/autoscaler/cluster-autoscaler/cloudprovider/gce	0.671s